src/datasets/translation_datasets.py

```diff
@@ -18,16 +23,19 @@ def __init__(
         self.lang2label: dict = {}
         self.label2lang: dict = {}
         label = 0
-        for pair in tqdm(self.pairs):
+        for pair in self.pairs:
             for lang in pair.split("-"):
                 if lang not in self.lang2label:
                     self.lang2label[lang] = label
                     self.label2lang[label] = lang
                     label += 1
-            dataset = load_dataset("allenai/nllb", pair, split="train", streaming=True)
-            self.datasets[pair] = iter(dataset)
+
             self.probas.append(probas[pair] ** k)
 
+        load_dataset("allenai/nllb", self.pairs[0], split="train", streaming=True)
```

Comment:
`load_dataset` dynamically creates a new module named  `datasets_modules` in the driver process. but there is some problems with multiprocessing with `load_dataset `function: by some reasons multiprocessing pool map won't create this module and thereby won't work. loading dataset of one pair is one of the solving problem.